app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt

```diff
@@ -1591,6 +1603,322 @@ class StateFragmentTest {
+      startPlayingExploration()
+
+      mergeDragAndDropItems(position = 0)
+
+      // Rotating device.
+      rotateToLandscape()
+      scrollToViewType(DRAG_DROP_SORT_INTERACTION)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.drag_drop_interaction_recycler_view,
+          position = 0,
+          targetViewId = R.id.drag_drop_item_recyclerview
+        )
+      ).check(matches(hasChildCount(2)))
+    }
+  }
+
+  @Test // TODO(#4692): Robolectric tests not working on screen rotation for input interactions
+  @RunOn(TestPlatform.ESPRESSO)
+  fun testStateFragment_sameTextBasedInteractions_doesNotShareInitialState() {
+    launchForExploration(TEST_EXPLORATION_ID_5, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+
+      typeNumericExpression("1+2")
+      rotateToLandscape()
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      it.onActivity {
+        val mathExpressionInteractionView =
+          it.findViewById<MathExpressionInteractionsView>(
+            R.id.math_expression_input_interaction_view
+          )
+        assertThat(mathExpressionInteractionView.text.toString().isEmpty()).isTrue()
+      }
+    }
+  }
+
+  @Test // TODO(#4692): Robolectric tests not working on screen rotation for input interactions
+  @RunOn(TestPlatform.ESPRESSO)
+  fun testStateFragment_differentTextBaseInteractions_doesNotShareInitialState() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+      // Entering text in Fraction Input Interaction
+      typeFractionText("1/2")
+      // Rotating device
+      rotateToLandscape()
+      it.onActivity {
+        val fractionInputInteraction =
+          it.findViewById<FractionInputInteractionView>(R.id.fraction_input_interaction_view)
+        assertThat(fractionInputInteraction.text.toString()).isEqualTo("1/2")
+      }
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+      playThroughPrototypeState3()
+      playThroughPrototypeState4()
+      playThroughPrototypeState5()
+
+      it.onActivity {
+        val numericInputInteractionView =
+          it.findViewById<NumericInputInteractionView>(R.id.numeric_input_interaction_view)
+        assertThat(numericInputInteractionView.text.toString().isEmpty()).isTrue()
+      }
+    }
+  }
+
+  @Test // TODO(#4692): Robolectric tests not working on screen rotation for input interactions
+  @RunOn(TestPlatform.ESPRESSO)
+  fun testStateFragment_checkPreviousHeader_retainStateOnConfigChange() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      // Attempt to submit an wrong answer.
+      typeFractionText("1/4")
+      clickSubmitAnswerButton()
+
+      // Attempt to submit an wrong answer.
```

Comment:
```suggestion
      // Attempt to submit a second wrong answer.
```

(Just to make it clear that the duplication is intentional).

Reply:
Updated